src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/SocketOutput.cs

```diff
@@ -694,40 +697,55 @@ private void PoolWriteReq(UvWriteReq writeReq)
 
             private void ScheduleReturnWrittenBlocks()
             {
-                Self._threadPool.UnsafeRun(_returnWrittenBlocks, this);
-            }
-
-            private void ReturnWrittenBlocks()
-            {
-                var block = _lockedStart.Block;
-                while (block != _lockedEnd.Block)
-                {
-                    var returnBlock = block;
-                    block = block.Next;
-
-                    returnBlock.Pool.Return(returnBlock);
-                }
+                var returnAll = false;
 
                 lock (Self._returnLock)
                 {
-                    // If everything has been fully written, return _tail.
+                    // If everything has been fully written, return _tail/_lockeEnd.
```

Comment:
You got eagle eyes! At this point you know to expect at least one spelling error in my PRs. 😝